ha/k8s_setup/ha_setup.py

```diff
@@ -370,15 +372,26 @@ def _add_cluster_component_health(self) -> None:
                                    specific_info=specific_info)
 
 
-    def _add_node_health(self, data_node_ids, server_node_ids, control_node_ids) -> None:
+    def _add_node_health(self, data_node_ids, server_node_ids, control_node_ids, kv_enable_batch_put: bool = True) -> None:
         """
         Add node health
         """
         _, _, node_mapping = self._confStoreAPI.get_cluster_cardinality()
+        node_id_list = []
+        node_id_len = len(node_mapping)
         # cluster cardinality node_to_name mapping will have actual machine-ids
         # If that is part of one of the list from data, server or control node
         # list, then mark the respective functional type and update the specific info
         for node_id in node_mapping.values():
+            node_id_list.append(node_id)
+            if len(node_id_list) > CONSUL_TRANSACTIONS_LIMIT:
+                # Note: if batch put is enabled, need to commit
+                # to push all the local cashed values to consul server
+                if kv_enable_batch_put:
+                    Log.debug(f"pushing {CONSUL_TRANSACTIONS_LIMIT} node health transactions to store")
+                    self._confstore.commit()
```

Comment:
@Madhura-08 I am not sure if we are setting cardinality for subsequent batches after first batch. Unless we do that it seems that cardinality set earlier will be reused. If I am missing any internal way to populate payload please correct me.

Reply:
Cardinality will be stored in the first iteration itself. For cardinality, we have only one key. Then this stored cardinality will be reused here to get the machine-ids out of it. And then these machine-ids will be used to create node health keys.
Ex: 
cortx>ha>v2>cluster_cardinality:{"num_nodes": 10, "node_id_dict": {"cortx-data-g0-2.cortx-data-headless.cortx.svc.cluster.local": "6ac2b08cfe659acf910fed4d11b7df53"}}

So, In this key, **6ac2b08cfe659acf910fed4d11b7df53** is the machine-id. We will iterate this node_id_dict for its values and then the extracted machine id will be used to create the node health key.
Ex node health key:
**cortx>ha>v2>cortx>ha>system>cluster>b59c6099-0f46-4e31-9215-aad37bf838e1>site>NOT_DEFINED>rack>NOT_DEFINED>node>f488442b875fe8c4328e148f8aca7aa7>health**

I hope this will help.

Reply:
I understand, my question was more towards setting payload for batch commit. Setting cardinality seems to be the only way as per API.